src/main/java/core/basesyntax/services/impl/ShopServiceImpl.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax.services.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.services.ShopService;
+import core.basesyntax.strategy.OperationStrategy;
+import java.util.List;
+
+public class ShopServiceImpl implements ShopService {
+    private final OperationStrategy operationStrategy;
+
+    public ShopServiceImpl(OperationStrategy operationStrategy) {
+        this.operationStrategy = operationStrategy;
+    }
+
+    @Override
+    public void process(List<FruitTransaction> transactions) {
+        for (FruitTransaction fruitTransaction: transactions) {
+            operationStrategy.getOperation(fruitTransaction.getOperation())
+                    .process(fruitTransaction);
+        }
+    }
```

Comment:
The class 'ShopServiceImpl' is correctly using dependency injection for the 'OperationStrategy'. However, ensure that this service is called in the 'main()' method of the 'Main' class to follow the task requirement of making services independent and orchestrating them in the 'main()' method.